tests/unit_tests/test_matchmaker_algorithm_team_matchmaker.py

```diff
@@ -0,0 +1,248 @@
+import logging
+
+import pytest
+
+from server import config
+from server.matchmaker import CombinedSearch, Search
+from server.matchmaker.algorithm.team_matchmaker import (
+    TeamMatchMaker,
+    UnevenTeamsException
+)
+
+
+@pytest.fixture(scope="module")
+def player_factory(player_factory):
+    player_id_counter = 0
+
+    def make(
+        mean: int = 1500,
+        deviation: int = 500,
+        ladder_games: int = config.NEWBIE_MIN_GAMES+1,
+        name=None
+    ):
+        nonlocal player_id_counter
+        """Make a player with the given ratings"""
+        player = player_factory(
+            ladder_rating=(mean, deviation),
+            ladder_games=ladder_games,
+            login=name,
+            lobby_connection_spec=None,
+            player_id=player_id_counter
+        )
+        player_id_counter += 1
+        return player
+    return make
+
+
+matchmaker = TeamMatchMaker(4)
```

Comment:
Well you don't want to make it a global variable because that could cause cross-test interference. I would make one per test, or you could make it a fixture in which case pytest will take care of making 1 per test for you.